Mathlib/RingTheory/DedekindDomain/PID.lean

```diff
@@ -191,8 +191,8 @@ theorem IsPrincipalIdealRing.of_finite_primes [IsDomain R] [IsDedekindDomain R]
     · exact isUnit_of_mul_eq_one _ _ (FractionalIdeal.coe_ideal_mul_inv I hI)⟩
 #align is_principal_ideal_ring.of_finite_primes IsPrincipalIdealRing.of_finite_primes
 
-variable [IsDomain R] [IsDedekindDomain R]
-variable (S : Type*) [CommRing S] [IsDomain S]
+variable [IsDedekindDomain R]
+variable (S : Type*) [CommRing S]
```

Comment:
Why did you decide to move the hypothesis `[IsDomain S]` to the statements of the results later on rather than keeping it here?

Reply:
It was used only in two results and I find it a bit overkilling to have it as a hanging hypothesis.